packages/front-end/components/Features/ValueDisplay.tsx

```diff
@@ -85,37 +89,89 @@ export default function ValueDisplay({
 
   return (
     <>
-      <Flex align="end" gap="2">
-        <Box flexGrow="1" style={fullStyle} className={fullClassName}>
+      <div style={{ position: "relative" }}>
+        <div style={fullStyle} className={fullClassName}>
           <InlineCode language="json" code={formatted} />
-        </Box>
-        {showFullscreenButton && type === "json" && (
-          <Flex align="end">
-            <Tooltip body="View in full screen" style={{ marginBottom: -6 }}>
-              <IconButton
-                radius="full"
-                variant="ghost"
-                onClick={() => setModalOpen(true)}
-              >
-                <PiCornersOutBold size={15} />
-              </IconButton>
-            </Tooltip>
-          </Flex>
+        </div>
+        {!isFullscreen && (
+          <div
+            className="d-flex"
+            style={{
+              position: "absolute",
+              bottom: -4,
+              right: 16,
+              gap: "0.75rem",
+            }}
+          >
+            {type === "json" || type === "string" ? (
+              <Tooltip body={copySuccess ? "Copied" : "Copy to clipboard"}>
+                <IconButton
+                  type="button"
+                  radius="full"
+                  variant="ghost"
+                  onClick={(e) => {
+                    e.preventDefault();
+                    e.stopPropagation();
+                    if (!copySuccess) performCopy(value);
+                  }}
+                >
+                  {copySuccess ? <PiCheck size={12} /> : <PiCopy size={12} />}
+                </IconButton>
+              </Tooltip>
+            ) : null}
+            {showFullscreenButton && type === "json" && (
+              <Tooltip body="View in full screen">
+                <IconButton
+                  type="button"
+                  radius="full"
+                  variant="ghost"
+                  onClick={(e) => {
+                    e.preventDefault();
+                    e.stopPropagation();
+                    setModalOpen(true);
+                  }}
+                >
+                  <PiCornersOut size={12} />
+                </IconButton>
+              </Tooltip>
+            )}
+          </div>
         )}
-      </Flex>
+      </div>
       {modalOpen && (
         <Modal
           header="Feature Value"
           open={modalOpen}
           close={() => setModalOpen(false)}
           trackingEventModalType=""
           size="max"
+          sizeY="max"
+          secondaryCTA={
+            copySuccess ? (
+              <Button style={{ width: 100 }} icon={<PiCheck />} color="gray">
+                Copied
+              </Button>
+            ) : (
+              <Button
+                style={{ width: 100 }}
+                icon={<PiCopy />}
+                onClick={() => {
+                  if (!copySuccess) performCopy(value);
```

Comment:
nit: Should we have `handleCopyButtonClick` and use for both fullscreen and not-fullscreen values?

Reply:
will leave it alone since there is slightly different behavior with event propagation